torch/distributed/elastic/rendezvous/c10d_rendezvous_backend.py

```diff
@@ -0,0 +1,195 @@
+# Copyright (c) Facebook, Inc. and its affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+import binascii
+import codecs
+import logging
+import os
+from datetime import timedelta
+from typing import Any, Optional, Tuple, cast
+
+from torch.distributed import Store, TCPStore
+
+from .api import RendezvousConnectionError, RendezvousParameters, RendezvousStateError
+from .default_rendezvous import RendezvousBackend, Token
+from .utils import _matches_machine_hostname, _parse_rendezvous_endpoint
+
+log = logging.getLogger(__name__)
+
+
+class C10dRendezvousBackend(RendezvousBackend):
+    """Represents a C10d-backed rendezvous backend.
+
+    Args:
+        store:
+            The `Store` instance to use to communicate with the C10d store.
+        run_id:
+            The run id of the rendezvous.
+    """
+
+    # See the explanation in the __init__ method.
+    _NULL_SENTINEL = "Y2FuaW1hZGFt"
+
+    _store: Store
+    _key: str
+
+    def __init__(self, store: Store, run_id: str) -> None:
+        if not run_id:
+            raise ValueError("The run id must be a non-empty string.")
+
+        self._store = store
+
+        self._key = "torch.rendezvous." + run_id
+
+        # The read operation of a store blocks the caller until the specified
+        # key becomes available. This behavior makes it tricky to use a store
+        # as a regular key-value dictionary.
+        #
+        # As a workaround we initially set a sentinel value as the rendezvous
+        # state. Whenever this value gets returned we treat it as a None.
+        self._call_store("compare_set", self._key, "", self._NULL_SENTINEL)
```

Comment:
Agree with changing to "set" as @tierex mentioned. Then we also wouldn't need to change compareSetHandler in TCPStore.cpp

Reply:
Please see my answer to him in Phabricator.